src/ties_remove_dangling.lua

```diff
@@ -2,18 +2,40 @@ function plugindef()
   -- This function and the 'finaleplugin' namespace
   -- are both reserved for the plug-in definition.
   finaleplugin.Author = "Jacob Winkler"
-  finaleplugin.Copyright = "2022"
-  finaleplugin.Version = "1.0.1"
-  finaleplugin.Date = "2022-08-26"
-  finaleplugin.RequireSelection = true
+  finaleplugin.Copyright = "2024"
+  finaleplugin.Version = "1.2&"
+  finaleplugin.Date = "2024-02-06"
   finaleplugin.AuthorEmail = "jacob.winkler@mac.com"
+  finaleplugin.AdditionalMenuOptions = [[
+    Ties: Replace Dangling w/Slurs
+    ]]
+  finaleplugin.AdditionalUndoText = [[
+    Ties: Replace Dangling w/Slurs
+    ]]
+  finaleplugin.AdditionalDescriptions = [[
+    Removes dangling ties and replaces them with slurs
+    ]]
+  finaleplugin.AdditionalPrefixes = [[
+    replace_with_slur = true
+    ]]
+  finaleplugin.Notes = [[
+      TIES: REMOVE DANGLING
+      
+      This script will search for 'dangling ties' - ties that go to rests rather than other notes - and either remove them or replace them with slurs.
+      ]]
   return "Ties: Remove Dangling", "Ties: Remove Dangling", "Removes dangling ties (ties that go nowhere)."
 end
 
+replace_with_slur = replace_with_slur or false
+
 local tie = require("library.tie")
+local smartshape = require("library.smartshape")
 
 local music_region = finale.FCMusicRegion()
 music_region:SetCurrentSelection()
+if music_region:IsEmpty() then
+  music_region:SetFullDocument()
+end
```

Comment:
**suggestion:** If you're going this route, then I would suggest using the library function, but it's fine if you want to leave this as-is.

```lua
local library = require("library.general_library")
local music_region = library.get_selected_region_or_whole_doc()
```